pkgs/applications/virtualization/singularity/generic.nix

```diff
@@ -177,9 +173,7 @@ in
     privileged-un-utils
     squashfsTools # mksquashfs unsquashfs # Make / unpack squashfs image
     squashfuse # squashfuse_ll squashfuse # Mount (without unpacking) a squashfs image without privileges
-  ]
-  ++ lib.optional enableNvidiaContainerCli nvidia-docker
-  ;
+  ];
```

Comment:
I believe `singularity-ce` still requires `libnvidia-container`? Or at least one of them does

Reply:
Unless I'm mistaken I think we can virtually get rid of `libnvidia-container` at some point. Singularity 4.0 supports CDI as well: https://sylabs.io/2023/10/introducing-cdi-support-to-singularityce-4-0/.

Do you know if `libnvidia-container` serves another usage not covered by CDI? I do wonder if there are still use cases depending on it, even if we introduce CDI.

I think if users want to use GPU's from Kubernetes the story is a bit different, have to check, but eventually everything will be CDI based as far as I know.

Reply:
Getting rid of everything we can would be great. Just to make sure, note that libnvidia-container is still one of the (optional?) backends for nvidia-container-toolkit, and also that apptainer and singularity-ce are separate projects which may diverge.

Reply:
There is some information at:

- https://github.com/NVIDIA/nvidia-docker/issues/1268#issuecomment-632692949 [1]
- https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/arch-overview.html [2]

As stated in [2]:

> In the past the nvidia-docker2 and nvidia-container-runtime packages were also discussed as part of the NVIDIA container stack. These packages should be considered deprecated as their functionality has been merged with the nvidia-container-toolkit package. The packages may still be available to introduce dependencies on nvidia-container-toolkit and ensure that older workflows continue to function. For more information on these packages, see the documentation archive for version older than v1.12.0

However, this wording relates to the fact that nvidia-container-runtime was an independent project (https://github.com/NVIDIA/nvidia-container-runtime) and that now it's been merged into the nvidia-container-toolkit. My understanding is that it will be completely replaced by CDI, but I have asked at https://github.com/NVIDIA/nvidia-container-toolkit/issues/335, just to have a clear understanding on that.

Reply:
There is also https://forums.developer.nvidia.com/t/what-is-unique-about-the-nvidia-container-runtime-does-cdi-replace-it/265975/5, that points on the CDI completely replaces nvidia-container-runtime direction too.

Reply:
I believe these two packages are distinct from `libnvidia-container`

Reply:
Brought libnvidia-container back